recipes/foxglove-websocket/all/conanfile.py

```diff
@@ -11,30 +17,98 @@ class FoxgloveWebSocketConan(ConanFile):
-    _source_package_path = os.path.join(_source_root, "cpp", "foxglove-websocket")
+    @property
+    def _compilers_minimum_version(self):
+        return {
+            "Visual Studio": "16.9",
+            "msvc": "191",
+            "gcc": "9",
+            "clang": "9",
+            "apple-clang": "12",
+        }
+
+    @property
+    def _source_package_path(self):
+        return os.path.join(self.source_folder, "tarball", "cpp", "foxglove-websocket")
 
     def source(self):
-        tools.get(**self.conan_data["sources"][self.version], strip_root=True, destination=self._source_root)
+        tmp_folder = "tarball"
+        get(self, **self.conan_data["sources"][self.version], strip_root=True, destination=tmp_folder)
+        copy(self, "*", src=os.path.join(self.source_folder, tmp_folder, "cpp", "foxglove-websocket"), dst=self.source_folder)
+        copy(self, "LICENSE", src=tmp_folder, dst=self.source_folder)
+        rmdir(self, tmp_folder)
 
     def validate(self):
         if self.settings.compiler.get_safe("cppstd"):
-            tools.check_min_cppstd(self, "17")
-        if (self.settings.compiler == "gcc" or self.settings.compiler == "clang") and tools.Version(self.settings.compiler.version) <= 8:
-            raise ConanInvalidConfiguration("Compiler version is not supported, c++17 support is required")
-        if self.settings.compiler == "Visual Studio" and tools.Version(self.settings.compiler.version) <= "16.8":
-            raise ConanInvalidConfiguration("Compiler version is not supported, c++17 support is required")
+            check_min_cppstd(self, self._min_cppstd)
+        minimum_version = self._compilers_minimum_version.get(str(self.settings.compiler), False)
+        if minimum_version and Version(self.settings.compiler.version) < minimum_version:
+            raise ConanInvalidConfiguration(
+                f"{self.ref} requires C++{self._min_cppstd}, which your compiler does not support."
+            )
+
+    def requirements(self):
+        self.requires("nlohmann_json/3.10.5", transitive_headers=True)
+        self.requires("websocketpp/0.8.2")
+
+    def layout(self):
+        if self.version >= Version("1.0.0"):
+            cmake_layout(self, src_folder="src")
+
+    def generate(self):
+        if self.version >= Version("1.0.0"):
+            tc = CMakeToolchain(self)
+            tc.generate()
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
 
     def configure(self):
         self.options["websocketpp"].asio = "standalone"
 
+        if self.version < Version("1.0.0"):
+            # Header only library
+            self.options.rm_safe("shared")
+            self.options.rm_safe("fPIC")
+        elif self.options.shared:
+            self.options.rm_safe("fPIC")
+
+    def build(self):
+        if self.version >= Version("1.0.0"):
+            cmake = CMake(self)
+            cmake.configure()
+            cmake.build()
```

Comment:
Should we remove the `0.0.1` version?

Reply:
Not sure if I understand correctly, do you mean to remove the `0.0.1` version from this recipe completely?

Reply:
Yeah, I mean, we're skipping the `0.0.1` version from the `layout`, `generate`, `build`, etc. steps, so why keep it there? @uilianries what do you think? In my opinion, let's avoid that `if` conditions if we're not going to build `0.0.1` anymore.

Reply:
https://github.com/conan-io/conan-center-index/pull/16500#discussion_r1190002679